File: java-vertx/src/main/java/io/swagger/server/api/model/V3BulkDeparturesRequest.java
package io.swagger.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.server.api.model.V3StopDepartureRequest;
import java.util.ArrayList;
import java.util.List;
import org.threeten.bp.OffsetDateTime;
import io.vertx.core.json.JsonObject;
import io.vertx.codegen.annotations.DataObject;
/**
 * V3BulkDeparturesRequest
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaVertXServerCodegen", date = "2024-04-14T03:35:56.953023300+10:00[Australia/Sydney]")



public class V3BulkDeparturesRequest   {
  
  private List<V3StopDepartureRequest> requests = new ArrayList<V3StopDepartureRequest>();

  private OffsetDateTime dateUtc = null;

  private Boolean lookBackwards = null;

  private Boolean includeCancelled = null;

  private Boolean includeGeopath = null;

  /**
   * Gets or Sets expand
   */
  public enum ExpandEnum {
    ALL("All"),
    STOP("Stop"),
    ROUTE("Route"),
    RUN("Run"),
    DIRECTION("Direction"),
    DISRUPTION("Disruption"),
    VEHICLEDESCRIPTOR("VehicleDescriptor"),
    VEHICLEPOSITION("VehiclePosition"),
    NONE("None");

    private String value;

    ExpandEnum(String value) {
      this.value = value;
    }

    public String getValue() {
      return value;
    }

    @Override
    public String toString() {
      return String.valueOf(value);
    }

    public static ExpandEnum fromValue(String value) {
      for (ExpandEnum b : ExpandEnum.values()) {
        if (b.value.equals(value)) {
          return b;
        }
      }
      return null;
    }

  }
  
  private List<ExpandEnum> expand = null;

  public V3BulkDeparturesRequest requests(List<V3StopDepartureRequest> requests) {
    this.requests = requests;
    return this;
  }

  public V3BulkDeparturesRequest addRequestsItem(V3StopDepartureRequest requestsItem) {
    this.requests.add(requestsItem);
    return this;
  }

  /**
   * Collection of departure requests
   * @return requests
   **/
    public List<V3StopDepartureRequest> getRequests() {
    return requests;
  }

  public void setRequests(List<V3StopDepartureRequest> requests) {
    this.requests = requests;
  }

  public V3BulkDeparturesRequest dateUtc(OffsetDateTime dateUtc) {
    this.dateUtc = dateUtc;
    return this;
  }

  /**
   * Filter by the date and time of the request (ISO 8601 UTC format) (default = current date and time)
   * @return dateUtc
   **/
    public OffsetDateTime getDateUtc() {
    return dateUtc;
  }

  public void setDateUtc(OffsetDateTime dateUtc) {
    this.dateUtc = dateUtc;
  }

  public V3BulkDeparturesRequest lookBackwards(Boolean lookBackwards) {
    this.lookBackwards = lookBackwards;
    return this;
  }

  /**
   * Indicates if filtering runs (and their departures) to those that arrive at destination before date_utc (default = false). Requires max_results &gt; 0.
   * @return lookBackwards
   **/
    public Boolean isLookBackwards() {
    return lookBackwards;
  }

  public void setLookBackwards(Boolean lookBackwards) {
    this.lookBackwards = lookBackwards;
  }

  public V3BulkDeparturesRequest includeCancelled(Boolean includeCancelled) {
    this.includeCancelled = includeCancelled;
    return this;
  }

  /**
   * Indicates if cancelled services (if they exist) are returned (default = false) - metropolitan train only
   * @return includeCancelled
   **/
    public Boolean isIncludeCancelled() {
    return includeCancelled;
  }

  public void setIncludeCancelled(Boolean includeCancelled) {
    this.includeCancelled = includeCancelled;
  }

  public V3BulkDeparturesRequest includeGeopath(Boolean includeGeopath) {
    this.includeGeopath = includeGeopath;
    return this;
  }

  /**
   * Indicates if the route geopath should be returned
   * @return includeGeopath
   **/
    public Boolean isIncludeGeopath() {
    return includeGeopath;
  }

  public void setIncludeGeopath(Boolean includeGeopath) {
    this.includeGeopath = includeGeopath;
  }

  public V3BulkDeparturesRequest expand(List<ExpandEnum> expand) {
    this.expand = expand;
    return this;
  }

  public V3BulkDeparturesRequest addExpandItem(ExpandEnum expandItem) {
    if (this.expand == null) {
      this.expand = new ArrayList<ExpandEnum>();
    }
    this.expand.add(expandItem);
    return this;
  }

  /**
   * List objects to be returned in full (i.e. expanded) - options include: all, stop, route, run, direction, disruption, none
   * @return expand
   **/
    public List<ExpandEnum> getExpand() {
    return expand;
  }

  public void setExpand(List<ExpandEnum> expand) {
    this.expand = expand;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3BulkDeparturesRequest v3BulkDeparturesRequest = (V3BulkDeparturesRequest) o;
    return Objects.equals(this.requests, v3BulkDeparturesRequest.requests) &&
        Objects.equals(this.dateUtc, v3BulkDeparturesRequest.dateUtc) &&
        Objects.equals(this.lookBackwards, v3BulkDeparturesRequest.lookBackwards) &&
        Objects.equals(this.includeCancelled, v3BulkDeparturesRequest.includeCancelled) &&
        Objects.equals(this.includeGeopath, v3BulkDeparturesRequest.includeGeopath) &&
        Objects.equals(this.expand, v3BulkDeparturesRequest.expand);
  }

  @Override
  public int hashCode() {
    return Objects.hash(requests, dateUtc, lookBackwards, includeCancelled, includeGeopath, expand);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3BulkDeparturesRequest {\n");
    
    sb.append("    requests: ").append(toIndentedString(requests)).append("\n");
    sb.append("    dateUtc: ").append(toIndentedString(dateUtc)).append("\n");
    sb.append("    lookBackwards: ").append(toIndentedString(lookBackwards)).append("\n");
    sb.append("    includeCancelled: ").append(toIndentedString(includeCancelled)).append("\n");
    sb.append("    includeGeopath: ").append(toIndentedString(includeGeopath)).append("\n");
    sb.append("    expand: ").append(toIndentedString(expand)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: csharp-dotnet2/src/main/CsharpDotNet2/IO/Swagger/Model/V3BulkDeparturesRequest.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class V3BulkDeparturesRequest {
    /// <summary>
    /// Collection of departure requests
    /// </summary>
    /// <value>Collection of departure requests</value>
    [DataMember(Name="requests", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "requests")]
    public List<V3StopDepartureRequest> Requests { get; set; }

    /// <summary>
    /// Filter by the date and time of the request (ISO 8601 UTC format) (default = current date and time)
    /// </summary>
    /// <value>Filter by the date and time of the request (ISO 8601 UTC format) (default = current date and time)</value>
    [DataMember(Name="date_utc", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "date_utc")]
    public DateTime? DateUtc { get; set; }

    /// <summary>
    /// Indicates if filtering runs (and their departures) to those that arrive at destination before date_utc (default = false). Requires max_results &gt; 0.
    /// </summary>
    /// <value>Indicates if filtering runs (and their departures) to those that arrive at destination before date_utc (default = false). Requires max_results &gt; 0.</value>
    [DataMember(Name="look_backwards", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "look_backwards")]
    public bool? LookBackwards { get; set; }

    /// <summary>
    /// Indicates if cancelled services (if they exist) are returned (default = false) - metropolitan train only
    /// </summary>
    /// <value>Indicates if cancelled services (if they exist) are returned (default = false) - metropolitan train only</value>
    [DataMember(Name="include_cancelled", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "include_cancelled")]
    public bool? IncludeCancelled { get; set; }

    /// <summary>
    /// Indicates if the route geopath should be returned
    /// </summary>
    /// <value>Indicates if the route geopath should be returned</value>
    [DataMember(Name="include_geopath", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "include_geopath")]
    public bool? IncludeGeopath { get; set; }

    /// <summary>
    /// List objects to be returned in full (i.e. expanded) - options include: all, stop, route, run, direction, disruption, none
    /// </summary>
    /// <value>List objects to be returned in full (i.e. expanded) - options include: all, stop, route, run, direction, disruption, none</value>
    [DataMember(Name="expand", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "expand")]
    public List<string> Expand { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class V3BulkDeparturesRequest {\n");
      sb.Append("  Requests: ").Append(Requests).Append("\n");
      sb.Append("  DateUtc: ").Append(DateUtc).Append("\n");
      sb.Append("  LookBackwards: ").Append(LookBackwards).Append("\n");
      sb.Append("  IncludeCancelled: ").Append(IncludeCancelled).Append("\n");
      sb.Append("  IncludeGeopath: ").Append(IncludeGeopath).Append("\n");
      sb.Append("  Expand: ").Append(Expand).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
